### PR TITLE
validate row prior to creating new line item, model changes to valida…

### PR DESCRIPTION
**Pull request by gpazo**
…te by Julian

### CODE REVIEW

File: lib/enyo-x/source/views/grid_box.js [hunk 1]
```diff
@@ -189,14 +189,20 @@ trailing:true, white:true, strict: false*/
...
+          this.doMoveDown(inEvent);
+          return true;
+        } else if (inEvent.keyCode === XV.KEY_UP) {
+          this.doMoveUp(inEvent);
+          return true;
+        } else if (inEvent.keyCode === XV.KEY_ENTER) {
+          this.doEnterOut(inEvent);
+          return true;
+        }
+        // Exit the event chain if req. attrs.
```
Comment (jcarlin) [inline lib/enyo-x/source/views/grid_box.js]: Oops l205, l206 should be moved to below/outside the `if`.

**Comment by Panduk** [comment]
Performed UAT: Issue is fixed.

File: lib/enyo-x/source/views/grid_box.js [hunk 2]
```diff
@@ -189,16 +189,22 @@ trailing:true, white:true, strict: false*/
...
-        return true;
-      } else if (inEvent.keyCode === XV.KEY_UP) {
-        this.doMoveUp(inEvent);
-        return true;
-      } else if (inEvent.keyCode === XV.KEY_ENTER) {
-        this.doEnterOut(inEvent);
-        return true;
+      var model = this.value;
+      // Don't navigate line items if the current model doesn't have all it's req. attrs.
+      if (!model.validate(model.attributes)) {
```
Comment (jcarlin) [inline lib/enyo-x/source/views/grid_box.js]: `if (!model.validate(model.attributes)) {`
change to
`if (model && !model.validate(model.attributes)) {`

File: lib/enyo-x/source/views/grid_box.js (hunk 1 above)
Comment (jcarlin) [inline lib/enyo-x/source/views/grid_box.js]: This comment should be moved to above the `return true;` on l207

**Pull request merged**